File: RedSheet.DbEntities/Models/ProjectPreparation.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("ProjectPreparations", Schema = "dbo")]
    [Serializable]
    public partial class ProjectPreparation
    {

        [Required]
        [MaxLength(400)]
        public string ElevatorSpeech { get; set; }

        [Required]
        [MaxLength(400)]
        public string PreConditioningMessage { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int ProjectPreparationId { get; set; }

        [RelationshipTableAttribue("ProjectModules", "dbo")]
        //Relationships
        public int ProjectModuleId { get; set; }

        [ForeignKey("ProjectModuleId")]
        public ProjectModule ProjectModule { get; set; }
        [InverseProperty("ProjectPreparation")]
        public ICollection<CommunicationPlan> CommunicationPlans { get; set; }

        [InverseProperty("ProjectPreparation")]
        public ICollection<EventPlanningAction> EventPlanningActions { get; set; }


        public ProjectPreparation()
        {
            this.CommunicationPlans = new HashSet<CommunicationPlan>();

            this.EventPlanningActions = new HashSet<EventPlanningAction>();

        }

    }
}

File: redsheet-portal/src/app/database-models/project-preparation.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, CommunicationPlan, EventPlanningAction, vProjectPreparationRecord } from './'
export class ProjectPreparation {
    constructor(projectPreparation?: ProjectPreparation | vProjectPreparationRecord) {
        let properties = ["elevatorSpeech", "preConditioningMessage", "projectPreparationId", "projectModuleId", "communicationPlans", "eventPlanningActions",];
        for (let property of properties)
            if (projectPreparation && projectPreparation[property])
                this[property] = projectPreparation[property];
    }

    @required()
    @maxLength(400)
    elevatorSpeech: string = undefined;

    @required()
    @maxLength(400)
    preConditioningMessage: string = undefined;

    projectPreparationId: number = 0;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;
    @nested(CommunicationPlan)
    communicationPlans: CommunicationPlan[];

    @nested(EventPlanningAction)
    eventPlanningActions: EventPlanningAction[];



}
